app/models/backend/transaction.rb

```diff
@@ -1,102 +1,43 @@
+# frozen_string_literal: true
+
+# Used by Stats tests and backend random data generator
 module Backend
-  class Transaction # this stuff is used by the stats tests, beware
-    attr_accessor :estimated_usage
-    attr_accessor :id
-    attr_accessor :provider_account_id
-    attr_accessor :created_at
-    #TODO: there's some duplication between this and delegate :to => :cinstance_data
-    attr_accessor :service_id
-    attr_accessor :client_ip
-    attr_accessor :cinstance_id
-    attr_accessor :log          # hash with the log info
+  class Transaction
+    attr_accessor :id, :estimated_usage, :provider_account_id, :created_at, :service_id, :client_ip, :cinstance_id, :log
 
     attr_writer :user_key
-    attr_writer :confirmed
 
-    delegate :provider_verification_key, :plan_name, :application_id,
-             :service_id, :to => :cinstance_data
+    delegate :provider_verification_key, :plan_name, :application_id, to: :cinstance_data
 
     def initialize(attributes = {})
-      @confirmed = false
-
       self.attributes = attributes
     end
 
-    def self.find(service_id, id)
-      find_by_id(service_id, id) || raise(TransactionNotFound)
-    end
-
-    def self.find_by_id(service_id, id)
-      attributes = storage.get_object(storage_key(service_id, id))
-      attributes && new(attributes.merge(:id => id))
-    end
-
     def self.report!(params = {})
       transaction = new(params)
       transaction.report!
       transaction
     end
 
-    # Report multiple transactions at the same time.
-    #
-    # Transaction.report_multiple!(@provider_account.id,
-    #                              '0' => {:user_key => @user_key_one,
-    #                                      :usage => {'hits' => 1},
-    #                                      :client_ip => "1.2.3.4"},
-    #                              '1' => {:user_key => @user_key_two,
-    #                                      :usage => {'hits' => 1},
-    #                                      :client_ip => "1.2.3.5"})
-    def self.report_multiple!(provider_account_id, raw_transactions)
-      reporter = MultiReporter.new(provider_account_id)
-      reporter.report!(raw_transactions)
-    end
-
     def self.usage_status(provider_account_id, cinstance_id, service_id, options = {})
       CinstanceData.new(provider_account_id, cinstance_id, service_id).status(options)
     end
 
     def report!
       self.created_at ||= Time.zone.now
 
-      if cinstance_data.anonymous_clients_allowed?
-        ensure_cinstance_exists
```

Comment:
This `ensure_cinstance_exists` method is missing, and I couldn't find anything about `anonymous_clients_allowed` feature, I think it's some legacy stuff that is not used anymore.